### PR TITLE
Jwt

### PR DESCRIPTION
**Pull request by vfilimo**

### CODE REVIEW

File: src/main/java/online/store/book/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    @Bean
+    public OpenAPI customOpenApi() {
+        return new OpenAPI()
+                .components(new Components().addSecuritySchemes("BearerAuth",
+                        new SecurityScheme()
+                                .type(SecurityScheme.Type.HTTP)
+                                .scheme("bearer")
+                                .bearerFormat("JWT")))
+                .addSecurityItem(new SecurityRequirement().addList("BearerAuth"));
```
Comment (MentorMateAcademy) [inline src/main/java/online/store/book/config/OpenApiConfig.java]: you can save strings to constants

File: src/main/java/online/store/book/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                            userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (MentorMateAcademy) [inline src/main/java/online/store/book/security/JwtAuthenticationFilter.java]: Save to constants

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/online/store/book/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                            userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HEADER_NAME);
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

use Spring constant

File: src/main/java/online/store/book/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String SECURITY_SCHEME_NAME = "Bearer ";
+    private static final String HEADER_NAME = "Authorization";
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/online/store/book/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String SECURITY_SCHEME_NAME = "Bearer ";
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/security/JwtAuthenticationFilter.java]:
```suggestion
    private static final String TOKEN_HEADER = "Bearer ";
```

File: src/main/java/online/store/book/security/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HEADER_NAME);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(SECURITY_SCHEME_NAME)) {
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(TOKEN_HEADER.length());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**